mr/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergWriteObjectInspector.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive.serde.objectinspector;
+
+public interface IcebergWriteObjectInspector {
+  Object getIcebergObject(Object value);
```

Comment:
We avoid `get` in APIs because it doesn't add much value and could be misleading in some cases. I think it is misleading here because this actually converts the object to the value to the Iceberg generic representation rather than just returning it. We also shouldn't need "Iceberg" in the name because it's in `IcebergWriteObjectInspector`.

How about naming this `convert`?

Also, where we have similar interfaces elsewhere, we tend to like to use parameterized types. [`ValueWriters`](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/avro/ValueWriters.java#L190-L200) is a good example.

Reply:
My reasoning was:
- This is an `ObjectInspector` only change and adds only a single new method to the Iceberg specific` ObjectInspector`s
- All of the `ObjectInspector` implementations are Overriding `get*` methods as inherited from Hive
- It might be reasonable to stick to one naming methods on Object level

That said, since I am not overly attached to the original naming schema, changed it to the suggested one.

Thanks,
Peter

Reply:
Oh.. and I plan to collapse the `DeserializerVisitor.primitive` to the following when we have implemented the handling of every Iceberg primitive type:
```
    @Override
    public FieldDeserializer primitive(PrimitiveType type, ObjectInspector inspector) {
      if (inspector instanceof WriteObjectInspector) {
        return o -> ((WriteObjectInspector) inspector).convert(o);
      } else {
        return o -> ((PrimitiveObjectInspector) inspector).getPrimitiveJavaObject(o);
      }
    }
```
Having differently named method for every type would make this impossible.